### PR TITLE
Fix build bdk-ffi for aarch64-apple-ios-sim

### PR DESCRIPTION
**Pull request by notmandatory**
The build-publish job is failing due to changes to the rustc `nightly` version, which breaks the `miniscript` build, see rust-bitcoin/rust-miniscript#541. I removed the `-Z build-std` part of the build step (it isn't needed anymore to build for the  [aarch64-apple-ios-sim](https://doc.rust-lang.org/nightly/rustc/platform-support.html) target), ~~this seems to fix the error for now~~ .. ~~nevermind it's still not working.~~, I also had to change the nightly version to an older version, I chose `nightly-2023-04-10` and that one works.

### CODE REVIEW

**Comment by notmandatory** [comment]
The workflow is working again, see test run in my repo here: https://github.com/notmandatory/bdk-swift/actions/runs/4834075893

File: .github/workflows/publish-spm.yaml
Comment (thunderbiscuit) [inline .github/workflows/publish-spm.yaml:63]: I don't want to hold up the release but can I ask why we're not using the `release-smaller` profile here?

**Review by thunderbiscuit — APPROVED** [review]
ADK 50bbb2847879c8ade685ad828e7cb85edbd79915. Just one small question, but feel free to merge as is.

File: .github/workflows/publish-spm.yaml
Comment (notmandatory) [reply .github/workflows/publish-spm.yaml:63]: Good suggestion, originally the release-smaller didn't work with nightly, but I just double check and it now seems to work. I'll update the workflow and retest it. If it works now will update this PR before merging it.

Comment (notmandatory) [reply .github/workflows/publish-spm.yaml]: btw, enabling the `release-smaller` profile for `aarch64-apple-ios-sim` makes the final xcframework zip file about 20MB smaller, and every little bit helps!

**Pull request merged**
